kafka/src/main/java/io/specmesh/kafka/Provisioner.java

```diff
@@ -97,54 +113,74 @@ public static int provisionTopics(final KafkaApiSpec apiSpec, final Admin adminC
      * @param apiSpec the api spec
      * @param baseResourcePath the path under which external schemas are stored.
      * @param schemaRegistryClient the client for the schema registry
+     * @return status of actions
      */
-    public static void provisionSchemas(
+    public static List<SchemaStatus> provisionSchemas(
             final KafkaApiSpec apiSpec,
             final String baseResourcePath,
             final SchemaRegistryClient schemaRegistryClient) {
 
         final var domainTopics = apiSpec.listDomainOwnedTopics();
 
+        final var statusList = new ArrayList<SchemaStatus>();
```

Comment:
I'd recommend refactoring to make this easier to grok:

```java
public static List<SchemaStatus> provisionSchemas(
            final KafkaApiSpec apiSpec,
            final String baseResourcePath,
            final SchemaRegistryClient schemaRegistryClient) {

      return apiSpec.listDomainOwnedTopics().stream()
              .map(domainTopic -> provisionSchema(domainTopic, apiSpec, baseResourcePath, schemaRegistryClient))
              .collect(toList());
 }


public static SchemaStatus provisionSchema(
            final String domainTopic,
            final KafkaApiSpec apiSpec,
            final String baseResourcePath,
            final SchemaRegistryClient schemaRegistryClient) {

    move code in lambda here.
    but move stuff into functions here too, e.g 

   try {
   final String schemaContent = readSchema(...). // throws specific exception
   registerSchema(...) // throws specific exception
   } catch (...) {
      status.exception(e);

      // Note: no throw here (which you currently have) as that will mean the returned list of statuses will stop at first error...
   }
   return status.build();
}
````

Reply:
yep - good call